Authorization.adoc

```diff
@@ -0,0 +1,71 @@
+= Authorization
+
+== Premise
+This document outlines a proposal for improving authorization in Infinispan.
+Currently, Infinispan authorization is disabled by default in both embedded and server.
+Authorization has two main scopes: global and per-cache. 
+Since per-cache operations authorization has a significant impact on performance, it needs to be explicitly enabled.
+
+== Current limitations
+* Configuration of authorization is complex and there is no OOTB configuration
+* The internal roles for schema and script management are not friendly
+* The small number of permissions means that the scope of the ADMIN permission is quite broad
+
+== Proposal
+The following changes are proposed:
+* Introduce a new CREATE permission
+* Include a default set of roles when the user doesn't provide any
+* Enable authorization out-of-the box in the server
+* (Optional) Introduce a way to "scope" permissions to resource names
+
+=== CREATE permission
+Add a CREATE permission to `org.infinispan.security.AuthorizationPermission`.
+CREATE allows creation/removal of caches, counters, schemas, scripts.
+CREATE supersedes and deprecates the existing `___schema_manager` and `___script_manager` internal roles.
+
+=== Default roles
+Enabling authorization without supplying any roles would define the following default roles:
+
+* *admin* superuser, allowed to do everything (*ALL*)
+* *application* allowed to perform all read/write ops, but not allowed to create/remove caches, schemas, scripts (*ALL_READ*, *ALL_WRITE*, *LISTEN*, *EXEC*)
+* *builder* allowed to create/remove caches, schemas, scripts (*ALL_READ*, *ALL_WRITE*, *LISTEN*, *EXEC*, *CREATE*)
```

Comment:
`LIFECYCLE` too?

Reply:
admin has ALL which implies LIFECYCLE. 
builder cannot stop the server

Reply:
so, the "builder" can create but cannot start a cache?

Reply:
You don't actually need LIFECYCLE to start a cache on the server... you just put the configuration in the `org.infinispan.CONFIG` cache, and the ConfigurationManager component creates+starts it.